src/execution/execute.ts

```diff
@@ -220,41 +215,32 @@ export function executeSync(args: ExecutionArgs): ExecutionResult {
   return result;
 }
 
-/**
- * Given a completed execution context and data, build the `{ errors, data }`
- * response defined by the "Response" section of the GraphQL specification.
- */
-function buildResponse(
-  data: ObjMap<unknown> | null,
-  errors: ReadonlyArray<GraphQLError>,
-): ExecutionResult {
-  return errors.length === 0 ? { data } : { errors, data };
+interface GraphQLExecutionPlanOptions {
+  fieldResolver?: Maybe<GraphQLFieldResolver<any, any>>;
+  typeResolver?: Maybe<GraphQLTypeResolver<any, any>>;
+  subscribeFieldResolver?: Maybe<GraphQLFieldResolver<any, any>>;
 }
 
-/**
- * Constructs a ExecutionContext object from the arguments passed to
- * execute, which we will pass throughout the other execution methods.
- *
- * Throws a GraphQLError if a valid execution context cannot be created.
- *
- * TODO: consider no longer exporting this function
- * @internal
- */
-export function buildExecutionContext(
-  args: ExecutionArgs,
-): ReadonlyArray<GraphQLError> | ExecutionContext {
-  const {
-    schema,
-    document,
-    rootValue,
-    contextValue,
-    variableValues: rawVariableValues,
-    operationName,
-    fieldResolver,
-    typeResolver,
-    subscribeFieldResolver,
-  } = args;
+interface ExecutionPlan {
+  schema: GraphQLSchema;
+  operation: OperationDefinitionNode;
+  fragments: ObjMap<FragmentDefinitionNode>;
+  rootType: GraphQLObjectType;
+  fieldResolver: GraphQLFieldResolver<any, any>;
+  typeResolver: GraphQLTypeResolver<any, any>;
+  subscribeFieldResolver: GraphQLFieldResolver<any, any>;
+}
 
+type ResultOrGraphQLErrors<T> =
```

Comment:
Is this helpful in terms of https://github.com/graphql/graphql-js/pull/3649 ?? -- I'm afraid I did not try to reproduce that error, and have not yet hit it with my TS version.

Reply:
Yes, exactly.
I missed #3649 but I always dislike this particular check.
I think we need to have a generic convention and this one seems like one that suits our use case the best since the error response is a valid GraphQL response.